File: syncloud/src/main/java/org/syncloud/android/Progress.java
package org.syncloud.android;

public interface Progress {
    void start();
    void stop();
    void error(String message);
    void title(String title);

    public class Empty implements Progress {

        @Override
        public void start() {

        }

        @Override
        public void stop() {

        }

        @Override
        public void error(String message) {

        }

        @Override
        public void title(String title) {

        }
    }
}

File: syncloud/src/main/java/org/syncloud/android/Progress.kt
package org.syncloud.android

interface Progress {
    fun start()
    fun stop()
    fun error(message: String?)
    fun title(title: String?)
    open class Empty : Progress {
        override fun start() {}
        override fun stop() {}
        override fun error(message: String?) {}
        override fun title(title: String?) {}
    }
}